internal/provider/resource_gitlab_group.go

```diff
@@ -369,9 +368,40 @@ func resourceGitlabGroupUpdate(ctx context.Context, d *schema.ResourceData, meta
 		return diag.FromErr(err)
 	}
 
+	if d.HasChange("parent_id") {
+		diagnostic := transferSubGroup(d, client)
+		if diagnostic.HasError() {
+			return diagnostic
+		}
+	}
+
 	return resourceGitlabGroupRead(ctx, d, meta)
 }
 
+func transferSubGroup(d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*gitlab.Client)
+	o, n := d.GetChange("parent_id")
+	parentId, ok := n.(int)
+	if !ok {
+		return diag.Errorf("error converting parent_id %v into an int", n)
+	}
+
+	opt := &gitlab.TransferSubGroupOptions{}
+	if parentId != 0 {
+		log.Printf("[DEBUG] transfer gitlab group %s from %v to new parent group %v", d.Id(), o, n)
+		opt.GroupID = gitlab.Int(parentId)
+	} else {
+		log.Printf("[DEBUG] turn gitlab group %s from %v to a new top-level group", d.Id(), o)
+	}
+
+	_, _, err := client.Groups.TransferSubGroup(d.Id(), opt)
```

Comment:
Let's use the context here from the CRUD function (has to be passed to the `transferSubGroup` function):

```suggestion
	_, _, err := client.Groups.TransferSubGroup(d.Id(), opt, gitlab.WithContext(ctx))
```

Reply:
completely agree, I forgot